CMR/python/cmr/search/granule.py

```diff
@@ -72,6 +73,57 @@ def granule_core_fields(item):
     record['native-id'] = meta.get('native-id')
     return {key: value for key, value in record.items() if value}
 
+def _collection_sample_limits(limit:int, climit:int, glimit:int):
+    """
+    Calculate out how many collections and granules should be used to satisfy
+    a requested sample. Allow for a simple case and a more defined case as such:
+
+    1. If a limit is provided, grab 10 granules from a set of collections
+
+    2. If no limit is given, then allow for a client and granule limit to be
+        specified and make assumptions if they are not.
+
+    """
+    if limit is None:
+        c_limit = 10 if climit is None else min(max(1, climit),100)
+        g_limit = 10 if glimit is None else min(max(1, glimit),100)
+        limit = c_limit * g_limit
+    else:
+        if glimit is None:
+            if climit is None:
+                c_limit = limit / 10
+                g_limit = 10
+            else:
+                c_limit = min(max(1, climit),100)
+                g_limit = limit / min(max(1, climit),100)
+        else:
+            #we have a limit and a glimit
+            if climit is None:
+                c_limit = limit / min(max(1, glimit),100)
+                g_limit = min(max(1, glimit),100)
+            else:
+                c_limit = min(max(1, climit),100)
+                g_limit = limit / min(max(1, climit),100)
+    return limit, math.ceil(c_limit), math.ceil(g_limit)
+
+def _collection_samples(collection_query, limit, config):
+    just_cid = lambda obj : obj.get('meta', {}).get('concept-id')
+    found_collections = scom.search_by_page("collections",
+        query=collection_query,
+        filters=just_cid,
+        page_state=scom.create_page_state(limit=limit),
+        config=config)
+    return found_collections[:limit]
+
+def _granule_samples(found_collections, filters, limit, g_limit, config):
+    found_granules = []
+    for concept in found_collections:
+        if len(found_granules) < limit:
+            query = {"concept_id": concept}
+            granules = search(query, filters=filters, limit=g_limit, config=config)
+            found_granules.extend(granules)
+    return found_granules[:limit]
+
```

Comment:
One can't tell by calling search here if it searches for granules or other things(I see search function specifically calls for granule search).  I think we should modify the search function to take in either "granules" or "collections", rather then calling search_by_page in _collection_samples, and calling search in _granule_samples.